metrics.nim

```diff
@@ -1048,35 +1046,65 @@ when defined(metrics) and defined(linux):
       # $ cat /proc/self/stat
       # 30494 (cat) R 3022 30494 3022 34830 30494 4210688 98 0 0 0 0 0 0 0 20 0 1 0 73800491 10379264 189 18446744073709551615 94060049248256 94060049282149 140735229395104 0 0 0 0 0 0 0 0 0 17 6 0 0 0 0 0 94060049300560 94060049302112 94060076990464 140735229397011 140735229397031 140735229397031 140735229403119 0
       let selfStat = readFile("/proc/self/stat").split(") ")[^1].split(' ')
-
-      process_virtual_memory_bytes.set(selfStat[20].parseFloat(), doUpdateSystemMetrics = false)
-      process_resident_memory_bytes.set(selfStat[21].parseFloat() * pagesize, doUpdateSystemMetrics = false)
-      process_start_time_seconds.set(selfStat[19].parseFloat() / ticks + btime, doUpdateSystemMetrics = false)
-      process_cpu_seconds_total.set((selfStat[11].parseFloat() + selfStat[12].parseFloat()) / ticks, doUpdateSystemMetrics = false)
+      result[@[]] = @[
+        Metric(
+          name: "process_virtual_memory_bytes", # Virtual memory size in bytes.
+          value: selfStat[20].parseFloat(),
```

Comment:
I see /proc/self/stat reads so this assumes Linux. In that case, isn't Rusage more standard? rather than parsing strings?

Reply:
see: https://github.com/mratsim/weave/blob/master/benchmarks/resources.nim

Reply:
Not in the Prometheus world. Their official Go client uses this: https://github.com/prometheus/procfs

And their official Python client does the same parsing we do: https://github.com/prometheus/client_python/blob/4e0e7ff7e615601a35376c414b7081945ed417f8/prometheus_client/process_collector.py#L62-L77